libs/config/node.go

```diff
@@ -0,0 +1,106 @@
+package config
+
+import "time"
+
+type Value struct {
+	v any
+	l Location
+
+	// Whether or not this value is an anchor.
+	// If this node doesn't map to a type, we don't need to warn about it.
+	anchor bool
+}
+
+var NilValue = Value{
+	v: nil,
+	l: Location{},
+}
+
+func New(v any, loc Location) Value {
```

Comment:
Maybe better call it `NewValue`? otherwise calls like `config.New` are a bit confusing.

Or even better it can be moved to `value` package and it can be called as `value.New`

Reply:
Also `config` is not a great name for this package..

I'll rename to `NewValue` though and we can figure out a better package name later.